website/src/audio-player.tsx

```diff
@@ -129,34 +137,119 @@ class AudioPlayer extends React.Component<
   }
 }
 
+const FunctionalAudioPlayer = (props: Props) => {
+  const [progress, setProgress] = useState(0)
+  const [howl, setHowl] = useState(
+    new Howl({
+      src: [props.audioUrl],
+      html5: true,
+      format: [".wav"],
+      preload: props.preload || false,
+      onplay: () => {
+        onPlay()
+      }, // What happens when a sound starts playing?
+      onend: () => {}, // What happens when a sound finishes playing?
+      onpause: () => {}, // What happens when a sound is paused?
+      onseek: () => {}, // What happens when a sound seeks?
+    })
+  )
+  const onPlay = () => {
+    requestAnimationFrame(nextStep)
+  }
+  const nextStep = () => {
+    let seek = howl.seek() || 0
+    setProgress((seek / howl.duration()) * 100)
+    if (howl.playing()) {
+      requestAnimationFrame(nextStep)
+    }
+  }
+
+  if (props.slices) {
+    let slices = {
+      __default: [props.slices.start, props.slices.end - props.slices.start],
+      sound: [props.slices.start, props.slices.end - props.slices.start],
+    }
+    Object.assign(howl, { _sprite: slices })
+  }
+
+  const onSeek = (newProgress: number) => {
+    setProgress(newProgress)
+    howl.seek((newProgress / 100) * howl.duration())
+  }
+
+  // Button set up
+  let button
+  if (progress > 0 && progress < 100 && howl.playing())
+    button = <PauseButton howl={howl} />
+  else button = <PlayButton howl={howl} isSprite={!!props.slices} />
+
+  let bounds = null
+  if (props.slices) {
+    bounds = {
+      start: props.slices.start / (howl.duration() * 10),
+      end: props.slices.end / (howl.duration() * 10),
+    }
+  }
+
+  // Finally return call
+  // Conditionally hide the bar
+  return (
+    <div className={css.audioElement}>
+      {button}
+      {props.showProgress && (
+        <ProgressBar progress={progress} seek={onSeek} bounds={bounds} />
+      )}
+    </div>
+  )
+}
+
+const ButtonSize = 28
```

Comment:
Only use UpperCamelCase for components, other constants should be lowerCamelCase, like `buttonSize`